astrapy/db.py

```diff
@@ -348,17 +348,26 @@ def get_collections(self):
         return response
 
     def create_collection(
-        self, dimension=None, options={}, function="", collection_name=""
+        self, options=None, dimension=None, metric="", collection_name=""
     ):
-        if dimension and not options:
-            options = {"vector": {"dimension": dimension}}
-            if function:
-                options["vector"]["function"] = function
-        if options:
-            jsondata = {"name": collection_name, "options": options}
-        else:
-            jsondata = {"name": collection_name}
+        # Initialize options if not passed
+        if not options:
+            options = {"vector": {}}
+        elif "vector" not in options:
+            options["vector"] = {}
+
+        # Now check the remaining parameters - dimension
+        if dimension and "dimension" not in options["vector"]:
+            options["vector"]["dimension"] = dimension
```

Comment:
nitpick: passing `options={... "vector": {"dimension": 123} ...}` AND `dimension=456` has the effect of silently discarding the 456 without warning the way I see it. Possibly dangerous?

Reply:
So i had this same thought, on the other hand, do we want to discard what would be in options? maybe we shouldnt allow `options` to be passed at all?